examples/siro_sandbox/controllers/controller_helper.py

```diff
@@ -97,23 +98,39 @@ def __init__(
                     gui_input=gui_input,
                 )
             self.controllers.append(gui_agent_controller)
-
             if is_multi_agent:
-                self.controllers.append(
-                    SingleAgentBaselinesController(
-                        0 if self._gui_controlled_agent_index == 1 else 1,
-                        is_multi_agent,
-                        config,
-                        self._gym_habitat_env,
+                if args.app_state == "fetch":
+                    self.controllers.append(
+                        FetchBaselinesController(
+                            0 if self._gui_controlled_agent_index == 1 else 1,
+                            is_multi_agent,
+                            config,
+                            self._gym_habitat_env,
+                        )
+                    )
+                else:
+                    self.controllers.append(
+                        SingleAgentBaselinesController(
+                            0 if self._gui_controlled_agent_index == 1 else 1,
+                            is_multi_agent,
+                            config,
+                            self._gym_habitat_env,
+                        )
                     )
-                )
 
     def get_gui_agent_controller(self) -> Optional[Controller]:
         if self._gui_controlled_agent_index is None:
             return None
 
         return self.controllers[0]
```

Comment:
I see we have the same sloppiness here as below (hard-coded magic numbers `0` and `1`). Can you just store `self._gui_controlled_agent_array_index` and `self._policy_controlled_agent_array_index` when these are constructed, and then leverage those indices in `get_gui_agent_controller` and `get_policy_driven_agent_controller`?

Reply:
Sorry that is an artifact from my policy eval fix and refactor. 

A bit of context:
We may have one or two controllers at SandboxApp runtime:
**Single agent case**
- SingleAgentBaselineController - one policy-controlled agent
- GuiController - one GUI-controlled agent

**Multi agent case**
- MultiAgentBaselineController - multiple policy-controlled agents
- [GuiController, SingleAgentBaselineController] - one GUI-controlled agent + one policy-controlled agent
- [GuiController, MultiAgentBaselineController] - one GUI-controlled agent + multiple policy-controlled agents (not supported yet)

Number of agents and number of controllers are not equal. We may have multiple agents, for example, [agent_1, agent_2, agent_3, agent_4] and we may set the agent_2 to be GUI controlled. As a result we sill have 2 controllers GuiController([agent_2]), MultiAgentBaselineController([agent_1, agent_3, agent_4]).

I see two possible solutions here:
1. As Eric suggested: (IIUC) set `self._gui_agent_controller_array_index` and `self._policy_driven_agents_controller_array_index` as we populate controllers list and reuse them in `get_gui_agent_controller` and `get_policy_driven_agent_controller`.
2. Since for all the possible cases above it will be enough to have up to 2 controllers to handle them. So maybe instead of storing controllers in the list, to avoid the confusion with the magic numbers, we should just explicitly define two attributes: `self._gui_agent_controller` and `self._policy_driven_agents_controller`? One of which may be set to None.

Option 1 looks better/more flexible to me.